packages/devtools_app/lib/src/flutter/table.dart

```diff
@@ -102,6 +114,17 @@ class _FlatTableState<T> extends State<FlatTable<T>> {
       backgroundColor: TableRow.colorFor(context, index),
     );
   }
+
+  void _sortDataAndUpdate(ColumnData column, SortDirection direction) {
+    setState(() {
+      sortData(column, direction);
+    });
+  }
+
+  @override
+  void sortData(ColumnData column, SortDirection direction) {
+    widget.data.sort((T a, T b) => _compareData(a, b, column, direction));
```

Comment:
we shouldn't be modifying data on a stateless widget. that is a violation of the implicit assumptions for how a stateless widget works. instead have the stateful widget create its own sorted copy.